sky/clouds/kubernetes.py

```diff
@@ -357,7 +357,7 @@ def query_status(cls, name: str, tag_filters: Dict[str, str],
         for pod in pods:
             if pod.status.phase == 'Running':
                 cluster_status.append(status_lib.ClusterStatus.UP)
-            elif pod.status.phase == 'Pending':
+            else:
                 cluster_status.append(status_lib.ClusterStatus.INIT)
         # If pods are not found, we don't add them to the return list
         return cluster_status
```

Comment:
Updated to return `INIT` in case it is not in a `Running` state. Otherwise, I think a failed pod is treated as not found and `sky status -r` will delete the cluster from local db.